### PR TITLE
New IO for NIX

### PR DESCRIPTION
**Pull request by achilleas-k**
This is a pull request to add an IO for [NIX](https://github.com/G-Node/nix). The IO uses the Python version of the library, [NIXPy](https://github.com/G-Node/nixpy) in pure Python mode (no compiled dependencies).

As of today, the version of NIXPy that is compatible with this IO should be on PyPI. The name of the package is `nixio` and the compatible version is `1.2.x`.

Note: NIX is a format based on HDF5 and uses the default file extension `.h5`. This may cause issues with file type detection if both the NIX IO and the HDF5 IO are used.

### CODE REVIEW

**Comment by apdavison** [comment]
The Travis failure is unlikely to have anything to do with this PR.

File: neo/io/nixio.py [hunk 1]
```diff
@@ -0,0 +1,1311 @@
...
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted under the terms of the BSD License. See
+# LICENSE file in the root of the Project.
+"""
+Module for reading data from files in the NIX format.
+
+Author: Achilleas Koutsou
+
+This IO supports both writina and reading of NIX files. Reading is supported
```
Comment (apdavison) [inline neo/io/nixio.py]: typo: writing

**Comment by achilleas-k** [comment]
I thought so too. It was strange. The 2.6 build was the only one that was failing before (due to unittest import) and after I fixed it, the 3.5 build failed.

**Comment by apdavison** [comment]
Many thanks for this @achilleas-k !

**Comment by achilleas-k** [comment]
You're very welcome.

**Pull request merged**